File: Semestrul 3/MAP/lab 06-07/src/main/java/ubb/scs/map/domain/entities/Entity.java
package ubb.scs.map.domain.entities;

public class Entity<ID>  {
    private ID id;
    /**
     * The id of the entity
     * @return the id of the entity
     */
    public ID getId() {
        return id;
    }
    /**
     * Sets the id of the entity
     * @param id the new id of the entity
     */
    public void setId(ID id) {
        this.id = id;
    }
}

File: Semestrul 4/MPP/lab02/cs/Domain/Entity.cs
namespace cs.Domain;
public abstract class Entity<ID>
{
    protected ID id;

    public Entity(ID id)
    {
        this.id = id;
    }

    public Entity()
    {
    }

    public ID GetId()
    {
        return id;
    }

    public void SetId(ID id)
    {
        this.id = id;
    }
}